hyperspy/learn/mva.py

```
@@ -958,6 +959,170 @@ def plot_explained_variance_ratio(self, n=50, log=True):
             ax.semilogy()
         return ax
 
+    def scree_plot(self, n=50, log=True, cutoff=0.01, signal_num=0,
```

I wouldn't change the name of the function. This is not exactly the scree plot as it doesn't plot the eigenvalues but the explained variance ratio. (And not even that when using poisson noise normalization...) So, why not simply replacing the old function?

---

I would set the `cutoff` to `None` by default as any uninformed guess will always be wrong and it can be misleading to the unaware user who may think that we have a clever way to do the job for him.

Talking about clever ways of estimating the number of PCs, maybe @tjof2 knows of any worth using? (I've tested the one in sklearn and it is so slow that I killed the computation before finding out if it was doing its job properly...)

---

You provide two features to highlight the signal components: different marker styles and the horizontal line. So, what about controlling them with the following keywords instead: `n_pcs=None`, `highlight_pcs=True`, `hline=True`?

---

@francisco-dlp not really for PCA - the reason I used soft rather than hard-thresholding (i.e. PCA) of the singular values was because it's easier to automate! @bm424 might know more.

---

> So, why not simply replacing the old function?

Sure, I can do that. I originally didn't want to be presumptuous and replace code :)

---

> as any uninformed guess will always be wrong and it can be misleading to the unaware user

This is a good point. Should we try to provide a good estimate, or just leave that up to the user? I found the following tutorial, which reviews a few different ways to determine the "right" number of components: http://eric.univ-lyon2.fr/~ricco/tanagra/fichiers/en_Tanagra_Nb_Components_PCA.pdf

---

It's notoriously hard to make a good automatic estimation so let's at least leave it for another PR.